### PR TITLE
initial implementation of publication to Zenodo

### PR DESCRIPTION
**Pull request by targos**

### CODE REVIEW

**Comment by targos** [comment]
Sorry it's not ready for review yet

**Comment by targos** [comment]
This is ready I think. There is nothing left specific to the ELN. I put options to allow for custom things

File: src/server/middleware/zenodo.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  sandbox: config.zenodoSandbox,
+  token: config.zenodoToken,
+  name: config.zenodoName,
+  visualizationUrl: config.zenodoVisualizationUrl,
+  readme: config.zenodoReadme,
+  attachments: config.zenodoAttachments
+});
+
+exports.createEntry = composeWithError(async (ctx) => {
+  const { entryId } = ctx.query;
```
Comment (stropitek) [inline src/server/middleware/zenodo.js]: readme says entry id is in the body

File: README.md [hunk 2]
```diff
@@ -125,6 +125,66 @@ Default: `false`
...
+This option is mandatory if `zenodo` is `true`.
+
+##### zenodoAttachments
+
+Type: function  
+Function that is called with each ROC entry's contents and must return a list of
+attachments to add to the Zenodo entry. It can also return an object if a single
+attachment is to be added
+
+```js
```
Comment (stropitek) [inline README.md]: Is this the default or just an example?

Comment (targos) [reply README.md]: Example. There is no default.

File: src/server/middleware/zenodo.js (hunk 1 above)
Comment (targos) [reply src/server/middleware/zenodo.js]: fixed readme

**Pull request merged**